site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/GroupController.java

```diff
@@ -290,8 +307,17 @@ public String saveGroup(@ModelAttribute GroupForm groupForm, Model model) {
                         addedGroupMemberList.add(member.getUserId());
                     }
                 }
-            // If the selected member is not a role, add it as individual member.
+            } else if (sectionList.stream().anyMatch(s -> selectedGroupMember.equals(s.getTitle()))) {
+                // If the selected member is a section, add all the members of this section to the group
+                selectedProviderIdList.add(selectedGroupMember);
+                Optional<Group> sectionOptional = sectionList.stream().filter(s -> selectedGroupMember.equals(s.getTitle())).findAny();
+                Group section = sectionOptional.get();
```

Comment:
The use of Optional.get() can produce a NoSuchElementException so it's best not to use this and use other variants of Optional, see https://www.baeldung.com/java-optional

Reply:
I know but there is an IF statement before that checks that

Reply:
see ifPresent or isPresent no need to call it twice

Reply:
good, replaced the logic using orElse